paasta_tools/tron_tools.py

```diff
@@ -364,6 +364,16 @@ def build_spark_config(self) -> Dict[str, str]:
             ),
             user=spark_tools.SPARK_JOB_USER,
         )
+        # delete the dynamically generated spark.app.id to prevent frequent config updates in Tron.
+        # spark.app.id will be generated later by yelp spark-submit wrapper or Spark itself.
+        spark_conf.pop("spark.app.id", None)
+        # use a static spark.app.name to prevent frequent config updates in Tron.
+        # md5 and base64 will always generate the same encoding for a string.
+        # This spark.app.name might be overridden by yelp spark-submit wrapper.
```

Comment:
from a sourcegraph search for `repo:^sysgit/yelpsoa-configs$ spark.app.name` it seems like some folks set short static app names, so we may want to follow this up with a follow-up PR that allows static/user-set app names to continue being set if that's something we want to support

Reply:
It's also auto set by spark-etl jobs - sourcegraph: `repo:spark_etl$ spark.app.name`

Reply:
Fixed this